### PR TITLE
Search: Show correct count for paginated results

### LINKED ISSUE #19656: "Found n Results" only counts first page of results
#### Steps to reproduce the issue

1. Start a search, e.g., https://jetpack.com/?s=backup
2. At first `jetpack-instant-search__search-results-title` will say `Found 10 results`
3. Scroll down to infinitely load the next page of results, and it will update to say `Found 20 results`

#### What I expected

The total number of results, regardless of how many are displayed on the current page. 

#### What happened instead

The count only includes results that have been loaded onto the page.

In my experience, that's not consistent w/ what search engines typically do. Seeing `10` made me think that there were _only_ 10 results.

In some cases, that could be confusing or misleading. e.g., if someone expects to see a couple dozen results, they may think that they searched for the wrong term and start trying others instead of scrolling to see all the results.

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #19656.

#### Changes proposed in this Pull Request:
* Shows correct count for paginated results.
  * Prior to this change, we would only rely on the number of results stored in our Redux state due to `response.total` being unreliable in the past (p8oabR-DS-p2#comment-5144).
  * With this change, we only override `response.total` if it's less than the number of results stored in our Redux state.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription. Ensure that Instant Search is enabled.
* Perform a search that yields more than 10 results. Ensure that the page says "Found n results" where n is the total count of all matching results.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js [hunk 1]
```diff
@@ -68,10 +68,10 @@ export function response( state = {}, action ) {
 				newState.results = [ ...( 'results' in state ? state.results : [] ), ...newState.results ];
 			}
 
-			// NOTE: There's a bug in the API where paginating to the next page can return an empty response.
-			//       This is most likely caused by setting excluded post types. Since `response.total` is unreliable
-			//       in this instance, manually set it to the length of the results array.
-			if ( Array.isArray( newState.results ) && newState.results.length !== newState.total ) {
+			// NOTE: There was a bug in the API where `response.total` was erroneously set to 0 when paginating
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js:71]: Do we need to mention the API bug any more? Could we just say:

"We override the total if the size of our results exceed the `response.total` value."

Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/store/reducer/api.js:71]: Good point, updated.

**Review by bluefuton — APPROVED** [review]
Looks good to me! ✅

**Review by jeherve — APPROVED** [review]

**Comment by jsnmoon** [comment]
🚢 r226874-wpcom

**Pull request merged**